docs: build kubevirt using local files (#8705)
kubevirt/kubevirt: docs/local-custom-files.md

# Build KubeVirt using local files

> KubeVirt system strongly relies on bazel and it is built mostly by files hosted remotely. In certain situations, it is handy to be able to build KubeVirt using custom local files. 

Reviewer: Maybe listing some of those "certain" situations would be helpful. Consider adding a couple of the most common reasons.

> The guide [custom-rpms.md](https://github.com/kubevirt/kubevirt/blob/main/docs/custom-rpms.md) already explains how to build KubeVirt using custom rpms, here, we specifically focus in using local files, but the 2 methods might be combined based on your needs.

Reviewer: ```suggestion
The guide [custom-rpms.md](https://github.com/kubevirt/kubevirt/blob/main/docs/custom-rpms.md) already explains how to build KubeVirt using custom rpms; while here we specifically focus in using local files, the 2 methods can be combined based on your needs.
```

> In the following example, we illustrate how to replace the `libguestfs-appliance` file, but it is avalid for any cases using remote files.

Reviewer 2: typo: `s/avalid/valid`

1. Copy your custom appliance file in the building container. It is enough to have the directory or the file in the kubevirt directory, and it will be automatically synchronized by the `hack/dockerized` command

```bash
# Local custom file
$ ls appliance/appliance-1.48.3.tar.xz 
appliance/appliance-1.48.3.tar.xz
# Sync build container and check the file 

> hack/dockerized ls appliance

Reviewer 2: ```suggestion
$ ./hack/dockerized ls appliance
```

go version go1.19.2 linux/amd64

go version go1.19.2 linux/amd64

> appliance-1.48.3.tar.xz

Reviewer 3: I think the second step should be `./hack/dockerized ls appliance/appliance-1.48.3.tar.xz` so that the results from outside of the container and inside it are more obviously comparable.

Maybe turn the `ls` into an `ls -l` as well, just to make the output stand out more? This one's arguably just a matter of taste :)

Author: I'll simply do an `ls` of the directory as all its content is copied inside the container
